### DISCUSSION TITLE
best way to upgrade airflow db from 2.3.2 to 2.5.1?

### DISCUSSION THREAD

**Discussion by amishpshah** [apache/airflow#30632] [General]
When I try to run the migrations to upgrade to 2.5.1, the migration times out. Once it times out, the database is only halfway migrated and then starts erroring out. What is the best way to upgrade the airflow to 2.5.1 from this old release?

**Comment by potiuk** [comment]
Strange but you might want to clean the db before migration to limit the amount of data to migrate . I think in 2.3.2 you already have `airflow db clean` command you can use to remove old data.

**Comment by kobethuwis** [comment]
Are you using the chart to do the migrations? Or are you running `airflow db upgrade` against the db directly?

**Comment by tdstark** [comment]
@amishpshah we ran into issues like this with a different upgrade. I'd suggest renaming your existing tables and initializing a new set of tables. You can then manually import and map the old data if you want to maintain your history. MySQL in particular is bad with DB migrations and upgrades.
